sdk/dotnet/Pulumi/Resources/StackReference.cs

```diff
@@ -0,0 +1,185 @@
+        public Output<object> RequireOutput(Input<string> name)
+        {
+            var inputs = (Input<ImmutableDictionary<string, object>>)this.Outputs;
+            var value = Output.Tuple(name, inputs).Apply(v => 
+                v.Item2.ContainsKey(v.Item1) 
+                    ? v.Item2[v.Item1]
+                    : throw new KeyNotFoundException(
+                        $"Required output '{name}' does not exist on stack '{Deployment.Instance.StackName}'."));
+
+            var isSecret = IsSecretOutputName(name);
+            return new Output<object>(value.Resources, value.ChangeSecretness(isSecret));
+        }
+
+        /// <summary>
+        /// Fetches the value of the named stack output. May return null if the value is
+        /// not known for some reason.
+        /// <para />
+        /// This operation is not supported (and will throw) for secret outputs.
+        /// </summary>
+        /// <param name="name">The name of the stack output to fetch.</param>
+        /// <returns>The value of the referenced stack output.</returns>
+        public async Task<object?> GetOutputValueAsync(Input<string> name)
+        {
+            var output = this.GetOutput(name);
+            var data = await output.DataTask.ConfigureAwait(false);
+            if (data.IsSecret)
+            {
+                throw new InvalidOperationException(
+                    "Cannot call 'GetOutputValueAsync' if the referenced stack has secret outputs. Use 'GetOutput' instead.");
+            }
+
+            return data.Value;
+        }
+
+        /// <summary>
+        /// Fetches the value promptly of the named stack output. Throws an error if the stack output is
+        /// not found.
+        /// <para />
+        /// This operation is not supported (and will throw) for secret outputs.
+        /// </summary>
+        /// <param name="name">The name of the stack output to fetch.</param>
+        /// <returns>The value of the referenced stack output.</returns>
+        public async Task<object> RequireOutputValueAsync(Input<string> name)
+        {
+            var output = this.RequireOutput(name);
+            var data = await output.DataTask.ConfigureAwait(false);
+            if (data.IsSecret)
+            {
+                throw new InvalidOperationException(
+                    "Cannot call 'RequireOutputValueAsync' if the referenced stack has secret outputs. Use 'RequireOutput' instead.");
+            }
+
+            return data.Value;
+        }
+
+        /// <summary>
+        /// Fetches the value of the named stack output as a secret, or null if the stack output was not found.
+        /// </summary>
+        /// <param name="name">The name of the stack output to fetch.</param>
+        /// <returns>An <see cref="Output{T}"/> containing the requested value.</returns>
+        public Output<object?> GetSecretOutput(Input<string> name)
```

Comment:
do we need this?  isn't this just waht GetOutput will always do?  is there a need to get an output from some other stack and reinterpret it as a secret?  do we have that in JS sdk?

Reply:
see the PR description with my question about that